proofread installation chapter (#2374)
centreon/centreon-documentation: versioned_docs/version-23.10/installation/installation-of-centreon-ha-manual/installation-2-nodes.md

---
id: centreon-ha-2-nodes-installation-manual-failover
title: 2-node manual failover installation
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Prerequisites

### Understanding

> Before you follow this procedure, we recommend that you have some familiarity with the Linux operating system and Centreon so you can understand what you are doing and correct any mistakes that might occur.

Reviewer: ```suggestion
Before you follow this procedure, we recommend that you have a good knowledge of the Linux operating system and Centreon so you can understand what you are doing and correct any mistakes that might occur.
```

CREATE USER '@MYSQL_REPL_USER@'@'@CENTRAL_SLAVE_IPADDR@' IDENTIFIED BY '@MYSQL_REPL_PASSWD@';
GRANT PROCESS, RELOAD, SHUTDOWN, SUPER, REPLICATION CLIENT, REPLICATION SLAVE ON *.* TO '@MYSQL_REPL_USER@'@'@DB_SLAVE_IPADDR@';

CREATE USER '@MYSQL_REPL_USER@'@'@CENTRAL_MASTER_IPADDR@' IDENTIFIED BY '@MYSQL_REPL_PASSWD@';
GRANT PROCESS, RELOAD, SHUTDOWN, SUPER, REPLICATION CLIENT, REPLICATION SLAVE ON *.* TO '@MYSQL_REPL_USER@'@'@DB_MASTER_IPADDR@';
```

### Configuring environment variables for MySQL scripts

The file `/etc/centreon-ha/mysql-resources.sh` contains environment variables that must be adapted to the current installation by replacing the macros with the appropriate value.

```bash
#!/bin/bash

###############################
# Database access credentials #
###############################

DBHOSTNAMEMASTER='@DB_MASTER_NAME@'
DBHOSTNAMESLAVE='@DB_SLAVE_NAME@'
DBREPLUSER='@MYSQL_REPL_USER@'
DBREPLPASSWORD='@MYSQL_REPL_PASSWD@'
DBROOTUSER='@MYSQL_REPL_USER@'
DBROOTPASSWORD='@MYSQL_REPL_PASSWD@'
CENTREON_DB='centreon'
CENTREON_STORAGE_DB='centreon_storage'

##########################################

> # Manual failover additional information #

Reviewer: ```suggestion
# Manual failover additional informations #
```

```bash
mysqladmin -p shutdown
```

Once the service is stopped on the **secondary** node, run the synchronization script **from the main node**: 

```bash
/usr/share/centreon-ha/bin/mysql-sync-bigdb.sh
```

This script does the following:

* checks that mysql is stopped on the secondary node
* stops mysql on the primary node
* mounts an LVM snapshot on the *volume group* that supports the /var/lib/mysql partition
* starts mysql on the primary node
* stores the current position in the binary logs
* disables the MySQL global variable `read_only` on the master node (*i.e.* the master node is now allowed to write to its database)
* synchronizes all data files (excluding the `mysql` system database) by overwriting the files on the secondary node
* unmounts the LVM snapshot
* creates the replication thread that will keep the data updated on the secondary node

This script is very verbose, and not everything that is displayed is understandable, but to make sure that it has run to completion, just make sure that the end looks like this:

```text

> Unmount and Delete LVM snapshot

Reviewer: ?

Start MySQL Slave
OK
Start Replication
Id	User	Host	db	Command	Time	State	Info	Progress
3	@MYSQL_REPL_USER@	@CENTRAL_MASTER_NAME@:33084	NULL	Query	0	init	show processlist	0.000
```

If all went well, then the `mysql-check-status.sh` command should return an error-free result:

```bash
/usr/share/centreon-ha/bin/mysql-check-status.sh
```

Expected outcome:

```text
Connection Status '@CENTRAL_MASTER_NAME@' [OK]
Connection Status '@CENTRAL_SLAVE_NAME@' [OK]
Slave Thread Status [OK]
Position Status [OK]
```

### Creating the MySQL VIP

The VIP will be added to the `@DB_VIP_IFNAME@` interface configuration on the master node:

```bash
nmcli con mod "@DB_VIP_IFNAME@" +ipv4.addresses "@DB_VIP_IPADDR@/@DB_VIP_CIDR_NETMASK@"
nmcli connection up "@DB_VIP_IFNAME@"
```

Then run the following script **on one of the two MySQL nodes** to check if the VIP is correctly mounted on the MySQL master server (this script can be run on either of the two nodes and it will detect by itself which is the master):

```bash
/usr/share/centreon-ha/bin/move-mysql-vip-to-mysql-master.sh
```

The result should be like this; if not, use the script to move the VIP to the correct place:

```text

> The VIP address is already in the right place. Nothing to do.

Reviewer: fixed in code

[Service]
EnvironmentFile=/etc/sysconfig/centreon_central_sync
ExecStart=/usr/share/centreon-ha/bin/centreon_central_sync $OPTIONS
ExecReload=/bin/kill -HUP $MAINPID
Type=simple
User=centreon

[Install]
WantedBy=centreon.service
EOF
```

#### Broker SQL service

```bash
cat > /usr/lib/systemd/system/cbd-sql.service <<"EOF"
[Unit]
Description=Centreon SQL Broker service
PartOf=centreon.service
ReloadPropagatedFrom=centreon.service
After=centreon-central-sync.service

[Service]
EnvironmentFile=/etc/sysconfig/cbd_sql
ExecStart=/usr/sbin/cbd $OPTIONS
ExecReload=/bin/kill -HUP $MAINPID
Type=simple
User=centreon-broker
UMask=0002

[Install]
WantedBy=centreon.service
EOF
```

#### Centengine service

```bash
cat > /usr/lib/systemd/system/centengine.service <<"EOF"
[Unit]
Description=Centreon Engine
PartOf=centreon.service
ReloadPropagatedFrom=centreon.service
After=cbd-sql.service

[Service]
ExecStart=/usr/sbin/centengine /etc/centreon-engine/centengine.cfg
ExecReload=/bin/kill -HUP $MAINPID
Type=simple
User=centreon-engine

[Install]
WantedBy=centreon.service
EOF
```

#### Centreontrapd service

```bash
cat > /usr/lib/systemd/system/centreontrapd.service <<"EOF"
[Unit]

> Description=Centreon Trapd Daemon is a Centreon program that manages traps

Reviewer: fixed in code
